src/api/materials.py

```
@@ -0,0 +1,541 @@
+        form_data = await req.formData()
+    except Exception:
+        return None, w.err("Failed to parse multipart form data", 400)
+
+    fields = {}
+    try:
+        # formData() returns a JS FormData object; iterate its entries.
+        for key, value in form_data.entries():
+            # File entries expose a .name attribute (the original filename).
+            filename = getattr(value, "name", None) or getattr(value, "filename", None)
+            if filename is not None:
+                # It's a File/Blob – read raw bytes.
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)